nsv13/code/modules/overmap/knpc.dm

```diff
@@ -253,35 +241,33 @@ GLOBAL_LIST_EMPTY(knpcs)
 	. = list()
 	var/mob/living/carbon/human/ai_boarder/H = HA.parent
 	for(var/mob/living/M in oview(HA.view_range, HA.parent))
-		//Invis is a no go. Dead mobs are ignored. Non-human, non hostile animals are ignored.
-		if(CHECK_BITFIELD(H.knpc_traits, KNPC_IS_MERCIFUL))
-			if(M.invisibility >= INVISIBILITY_ABSTRACT || M.alpha <= 0 || M.stat >= UNCONSCIOUS || (!ishuman(M) && !istype(M, /mob/living/simple_animal/hostile)))
-				continue
-		else
-			if(M.invisibility >= INVISIBILITY_ABSTRACT || M.alpha <= 0 || M.stat == DEAD || (!ishuman(M) && !istype(M, /mob/living/simple_animal/hostile)))
-				continue
+		//Invis is a no go. Non-human, non hostile animals are ignored.
+		if(M.invisibility >= INVISIBILITY_ABSTRACT || M.alpha <= 0 || (!ishuman(M) && !ishostile(M)))
+			continue
+		// Dead mobs are ignored.
+		if(CHECK_BITFIELD(H.knpc_traits, KNPC_IS_MERCIFUL) && M.stat >= UNCONSCIOUS)
+			continue
+		else if(M.stat == DEAD)
+			continue
 		if(H.faction_check_mob(M))
 			continue
-		if(!can_see(H, M, HA.view_range))
-			return
 		. += M
 	//Check for nearby mechas....
-	if(GLOB.mechas_list?.len)
-		for(var/obj/mecha/OM in GLOB.mechas_list)
+	if(length(GLOB.mechas_list))
+		for(var/obj/mecha/OM as() in GLOB.mechas_list)
```

Comment:
`as anything` just looks better

Reply:
`as anything` is needlessly longer, as() is generally more common on most codebases but ultimately it's down to preference.

Reply:
as() is less readable

Reply:
How?